timesketch/lib/google_jwt_test.py

```diff
@@ -0,0 +1,347 @@
+kf9K2hOUz1wkMPXPkTP04AkCgYAnPGDRibZ3rmGUwesMPeSJHMU3Gqr3csM5hXLk
+cwZ+xS/12sG6K4IApN6W/CJook81hTEjbx6svxfSeKYJHe9kjkjLlTMenW5WHl/R
+4dHTovwMvQCoMA0dyJ6rNI3XUKwmhO8cVigCxwz52g2K5LIVzRvINmKxqDI2x84c
+2WYPEQKBgAU26m/CfLEANjrN2gHVXaBA6uKnUnyQBl99PXH/HOJ7g8o8OXJVJIDe
+taxXqti9HRcj+oSgKNlL4A4W7pUmDh5sv6tHIx16MVoJO++2CjQN8V+m9DjPlgEx
++UVmw9npr/ilfWf3MnrUdl/MefdPUDZQ9rMIaDvX3VFA9H1KsJqv
+-----END RSA PRIVATE KEY-----
+"""
+
+IAP_VALID_AUDIENCE = '/projects/1234/global/backendServices/1234'
+IAP_INVALID_AUDIENCE = '/projects/foo/global/backendServices/bar'
+
+OIDC_CLIENT_ID = '123456'
+OIDC_VALID_AUDIENCE = OIDC_CLIENT_ID
+
+# Defaults for Google Cloud Identity-Aware Proxy
+IAP_PUBLIC_KEY_URL = 'https://www.gstatic.com/iap/verify/public_key'
+IAP_VALID_ISSUER = 'https://cloud.google.com/iap'
+IAP_JWT_ALGORITHM = 'ES256'
+
+# Defaults for Google OAuth/OpenID Connect
+OIDC_PUBLIC_KEY_URL = 'https://www.googleapis.com/oauth2/v3/certs'
+OIDC_VALID_ISSUER = 'https://accounts.google.com'
+OIDC_JWT_ALGORITHM = 'RS256'
+
+
+# pylint: disable=unused-argument
+def mock_fetch_iap_public_keys(unused_argument):
+    """Mock getting public keys.
+
+    Returns:
+        Mock public key.
+    """
+    return {"iap_1234": MOCK_EC_PUBLIC_KEY}
+
+
+# pylint: disable=unused-argument
+def mock_fetch_oidc_public_keys(unused_argument):
+    """Mock getting public keys.
+
+    Returns:
+        Mock public key.
+    """
+    return {"keys": [MOCK_RSA_PUBLIC_KEY_JWK]}
+
+
+def create_default_payload(audience, issuer):
+    now = int(time.time())
+    payload = {
+        "sub": "1234567890",
+        "email": "test@example.com",
+        "hd": "example.com",
+        "iat": now - 300,  # issued 5 min ago
+        "exp": now + 300,  # expires in 5 min
+        "aud": audience,
+        "iss": issuer
+    }
+    return payload
+
+
+def create_default_header(algorithm, key_id):
+    header = {
+        "alg": algorithm,
+        "kid": key_id
+    }
+    return header
+
+
+def create_mock_jwt(key, algorithm, key_id, audience, issuer, header=None,
+                    payload=None):
+    """Create test JSON Web Token (JWT).
+
+    Args:
+        key: Private key for signing.
+        algorithm: Algorithm used for the private key.
+        key_id: Key ID for the public key.
+        header: Dict with optional header.
+        payload: Dict with optional payload.
+
+    Returns:
+        Encoded JWT signed with key.
+    """
+    if not header:
+        header = create_default_header(algorithm=algorithm, key_id=key_id)
+
+    if not payload:
+        payload = create_default_payload(audience, issuer)
+
+    return jwt.encode(
+        payload, key, algorithm=algorithm, headers=header)
+
+
+@mock.patch(
+    u'timesketch.lib.google_jwt.fetch_public_keys', mock_fetch_iap_public_keys)
+class TestGoogleCloudIAP(BaseTest):
+    """Tests for the functionality of the google_jwt module."""
+
+    def _test_payload(self, payload, domain=None):
```

Comment:
I would rename this (and subsequent calls) to `_test_payload_raises_jwtvalidationerror`

Reply:
Done.